### PR TITLE
*: add TiDB roadmap

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add TiDB roadmap.

Preview link: https://pingcap-docs-website-scaffold-1hq.pages.dev/zh/tidb/dev/tidb-roadmap

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/13153
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-roadmap.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        </ul>
+      </td>
+      <td>
+        <ul>
+          <li>
+            <b>联邦查询</b>
+          </li>
+          <br />
+          <li>
+            <b>级联优化器 (Cascades optimizer)</b>
```
Comment (ran-huang) [inline tidb-roadmap.md:200]: 不确定术语

File: tidb-roadmap.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+      <td>
+        <ul>
+          <li>
+            <b>数据导入性能大幅提升</b>
+            <br /><i>预计提升 3~4 倍</i>
+          </li>
+          <br />
+          <li>
+            <b>TiCDC 支持多个上游数据源</b>
+            <br /><i>支持从多个 TiDB 集群到 TiCD (N:1)</i>
```
Comment (ran-huang) [inline tidb-roadmap.md:260]:
```suggestion
            <br /><i>支持从多个 TiDB 集群到 TiCDC (N:1)</i>
```

File: tidb-roadmap.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+            <b>升级期间自动暂停/恢复 DDL</b>
+            <br /><i>提供平滑的升级体验</i>
+          </li>
+          <br />
+        </ul>
+      </td>
+      <td>
+        <ul>
+          <li>
+            <b>AI 索引 (AI-indexing)</b>
```
Comment (ran-huang) [inline tidb-roadmap.md:280]: 不确定术语

File: tidb-roadmap.md [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+          </li>
+          <br />
+          <li>
+            <b>支持迁移异构数据库</b>
+          </li>
+          <br />
+          <li>
+            <b>使用 AI 赋能 SQL 性能优化 (Re-invented AI-SQL performance advisor)</b>
```
Comment (ran-huang) [inline tidb-roadmap.md:288]: 不确定术语

File: tidb-roadmap.md [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+      <td>
+        <b>数据库管理与可观测性</b>
+        <br /><i>增强数据库可管理性及其生态系统</i>
+      </td>
+      <td>
+        <ul>
+          <li>
+            <b>TiCDC 支持分布式单表复制</b>
+            <br /><i>
+              通过多个节点分布式执行工作负载，大幅提高 TiCDC 吞吐量
```
Comment (qiancai) [inline tidb-roadmap.md:224]:
```suggestion
              通过多个节点分布式执行同步负载，大幅提高 TiCDC 吞吐量
```

File: tidb-roadmap.md [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+        </ul>
+      </td>
+      <td>
+        <ul>
+          <li>
+            <b>联邦查询</b>
+          </li>
+          <br />
+          <li>
+            <b>级联优化器 (Cascades optimizer)</b>
```
Comment (ran-huang) [inline tidb-roadmap.md:178]:
```suggestion
            <b>Cascades 优化器</b>
```

File: tidb-roadmap.md [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+            <b>升级期间自动暂停/恢复 DDL</b>
+            <br />提供平滑的升级体验
+          </li>
+          <br />
+        </ul>
+      </td>
+      <td>
+        <ul>
+          <li>
+            <b>AI 索引 (AI-indexing)</b>
```
Comment (ran-huang) [inline tidb-roadmap.md:250]:
```suggestion
            <b>AI 索引</b>
```

File: tidb-roadmap.md [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+          </li>
+          <br />
+          <li>
+            <b>支持迁移异构数据库</b>
+          </li>
+          <br />
+          <li>
+            <b>使用 AI 赋能 SQL 性能优化 (Re-invented AI-SQL performance advisor)</b>
```
Comment (ran-huang) [inline tidb-roadmap.md:258]:
```suggestion
            <b>使用 AI 赋能 SQL 性能优化</b>
```

File: tidb-roadmap.md [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+        </ul>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+上述表格中并未列出所有功能，当前规划可能会调整。不同的服务订阅版本中的功能可能有所不同。
+
+## 近期已发布特性
+
```
Comment (ran-huang) [inline tidb-roadmap.md:326]:
```suggestion

- [TiDB 7.1.0 Release Notes](/releases/release-7.1.0.md)
```

**Comment by lilin90** [comment]
Note: We won't cherry pick this PR to other branches. Instead, @ran-huang will create another PR to add the roadmap URL (one URL similar to the English version https://docs.pingcap.com/tidb/dev/tidb-roadmap) to the `TOC.md` of necessary branches such as release-7.1.

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**